x/nft/model.go

```diff
@@ -0,0 +1,59 @@
+package nft
+
+import (
+	"github.com/iov-one/weave"
+	"github.com/iov-one/weave/orm"
+)
+
+var _ orm.CloneableData = (*NonFungibleToken)(nil)
+
+const (
+	minIDLength = 4
+	maxIDLength = 256
+)
+
+func (m *NonFungibleToken) Validate() error {
+	if len(m.Id) < minIDLength || len(m.Id) > maxIDLength {
+		return ErrInvalidID()
+	}
+	if err := weave.Address(m.Owner).Validate(); err != nil {
+		return err
+	}
+	// TODO: impl proper validation
```

Comment:
This is what #126 is about?

Reply:
Yes, so basically what I wanted to do there is to instantiate ops, pass the token "type",  which we don't have yet and be able to validate approvals fully in the base nft (here).

Alex's suggestion was to split approval validation in several parts and validate it part here and part in impl token.

Alex's approach seems good if we'd need some different logic there, which we don't atm as well as not knowing about type-specific action constants in Approval ops. My argument for my approach was that if we have this centralised - we can validate everything in base nft + make sure we throw an error if we don't know a type.